test/topology_custom/test_mv_topology_change.py

```diff
@@ -77,3 +81,87 @@ async def do_writes(start_it, repeat) -> int:
     stop_event.set()
     await asyncio.gather(*tasks)
 
+# Reproduces #19152
+# Verify a pending replica is not doing unnecessary work of building and sending view updates.
+# 1) we have a table with a materialized view with RF=1.
+#    the base and view tablets start on node 1.
+# 2) start migrating a base-table tablet from node 1 to node 2
+# 3) while node 2 is a pending replica, write to the table
+# 4) complete migration
+# 5) verify node 2 did not build view updates
+# With the parameter intranode=True it's the same except the tablet
+# is migrating between two shards on the same node.
+@pytest.mark.parametrize("intranode", [True, False])
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_mv_update_on_pending_replica(manager: ManagerClient, intranode):
+    cfg = {'enable_tablets': True}
+    cmd = ['--smp', '2']
+    servers = [await manager.server_add(config=cfg, cmdline=cmd)]
+
+    await manager.api.disable_tablet_balancing(servers[0].ip_addr)
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE test WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'initial': 1};")
+    await cql.run_async("CREATE TABLE test.test (pk int PRIMARY KEY, c int);")
+    await cql.run_async("CREATE MATERIALIZED VIEW test.mv1 AS SELECT * FROM test.test WHERE pk IS NOT NULL AND c IS NOT NULL PRIMARY KEY (c, pk);")
+
+    table_id = await manager.get_table_id('test', 'test')
+
+    servers.append(await manager.server_add(config=cfg, cmdline=cmd))
+
+    key = 7 # Whatever
+    tablet_token = 0 # Doesn't matter since there is one tablet
+    await cql.run_async(f"INSERT INTO test.test (pk, c) VALUES ({key}, 0)")
+
+    replica = await get_tablet_replica(manager, servers[0], 'test', 'test', tablet_token)
+    s0_host_id = await manager.get_host_id(servers[0].server_id)
+    s1_host_id = await manager.get_host_id(servers[1].server_id)
+    assert replica[0] == s0_host_id
+    assert replica[1] == 0
```

Comment:
How do we know that the single tablet must be on shard 0 and not on shard 1?
Come to think of it, how do you know the node is servers[0] and not servers[1]?

Reply:
I fixed the test to not assume the shard.
I think the tablet is on server 0 because we start with only server 0, create the tables and disable tablet balancing, and then start server 1.

Reply:
Thanks. I didn't see that you create the second server after having already created the view and disabling migration.